error caused by cpu-features.c getting compiled with CXX

### Crypto++ Issue Report



* OS:
  + Host: Xubuntu: Linux 5.3.0-26-generic #28~18.04.1-Ubuntu SMP Wed Dec 18 16:40:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
  + Target: Android 24, NDK 21.0.6113669
* Crypto++ 8.2.0
* Build script based on setenv-android.sh: [build_crypto++_android.sh](https://github.com/weidai11/cryptopp/files/4122880/build_crypto%2B%2B_android.sh.txt)
* Error:
```
armv7a-linux-androideabi24-clang++ -DNDEBUG -g2 -O3 -fPIC -pipe -Wall -march=armv7-a -mthumb -mfloat-abi=softfp -funwind-tables -fexceptions -frtti -DANDROID --sysroot=/home/paul/Android/Sdk/ndk/21.0.6113669//toolchains/llvm/prebuilt/linux-x86_64//sysroot -Wa,--noexecstack -I -I -fpermissive -c cpu-features.c
clang++: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
cpu-features.c:222:13: error: assigning to 'const char *' from incompatible type 'void *'
        p = memmem(p, bufend-p, field, fieldlen);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cpu-features.c:234:10: error: assigning to 'const char *' from incompatible type 'const void *'
    p  = memchr(p, ':', bufend-p);
         ^~~~~~~~~~~~~~~~~~~~~~~~
cpu-features.c:240:9: error: assigning to 'const char *' from incompatible type 'const void *'
    q = memchr(p, '\n', bufend-p);
        ^~~~~~~~~~~~~~~~~~~~~~~~~
cpu-features.c:246:14: error: assigning to 'char *' from incompatible type 'void *'
    result = malloc(len+1);
             ^~~~~~~~~~~~~
cpu-features.c:399:13: error: assigning to 'const char *' from incompatible type 'const void *'
        q = memchr(p, ',', end-p);
            ^~~~~~~~~~~~~~~~~~~~~
cpu-features.c:693:15: error: assigning to 'char *' from incompatible type 'void *'
    cpuinfo = malloc(cpuinfo_len);
```

* Fix: Changing compiler in GNUmakefile-cross from CXX to CC

--- a/GNUmakefile-cross
+++ b/GNUmakefile-cross
@@ -858,7 +858,7 @@ aes_armv4.o : aes_armv4.S
 	$(CXX) $(strip $(CXXFLAGS) $(CRYPTOGAMS_ARMV7_THUMB_FLAG) -c) $<
 
 cpu-features.o: cpu-features.h cpu-features.c
-	$(CXX) -x c $(strip $(CXXFLAGS) -c) cpu-features.c
+	$(CC) -x c $(strip $(CXXFLAGS) -c) cpu-features.c
 
 # SSSE3 or NEON available
 aria_simd.o : aria_simd.cpp

